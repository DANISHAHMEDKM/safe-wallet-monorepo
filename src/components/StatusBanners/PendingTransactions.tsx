import { View, Theme, Text, Circle } from 'tamagui'
import { SafeFontIcon } from '@/src/components/SafeFontIcon/SafeFontIcon'

import { ListItemTitle, ListItem } from 'tamagui'
import React from 'react'

interface Props {
  number: number
}

export const PendingTransactions = ({ number }: Props) => {
  const ArrowRight = <SafeFontIcon name="arrow-right" size={20} />

  return (
    <Theme name={'pendingTx'}>
      <View justifyContent={'center'} alignItems={'center'} flexDirection={'row'}>
        <View>
          <ListItem
            backgroundColor="$background"
            color="$color"
            borderRadius="$2"
            flexDirection="row" // Add this to ensure the items are in a row
            alignItems="center" // Ensures vertical centering
            icon={
              <Circle size={'$5'} backgroundColor={'$color'}>
                <Text color={'$badgeTextColor'}>{number}</Text>
              </Circle>
            }
            iconAfter={ArrowRight}
            size="$2"
            padding={'$2'}
            unstyled={false}
          >
            <ListItemTitle textAlign={'center'} fontWeight={'600'}>
              Pending Transactions
            </ListItemTitle>
          </ListItem>
        </View>
      </View>
    </Theme>
  )
}
