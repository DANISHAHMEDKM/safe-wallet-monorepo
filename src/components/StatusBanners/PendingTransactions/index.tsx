import { SafeFontIcon } from '@/src/components/SafeFontIcon/SafeFontIcon'

import React from 'react'
import { Alert } from '../../Alert'
import CircularBadge from '../../CircularBadge'
import { Spinner } from 'tamagui'

interface Props {
  number: string
  fullWidth?: boolean
  onPress: () => void
  isLoading?: boolean
}

const PendingTransactions = ({ number, isLoading, fullWidth, onPress }: Props) => {
  const startIcon = isLoading ? (
    <Spinner size="small" color="$warning1ContrastTextDark" />
  ) : (
    <CircularBadge content={number} />
  )
  const endIcon = <SafeFontIcon name="arrow-right" size={20} />

  return (
    <Alert
      type="warning"
      fullWidth={fullWidth}
      endIcon={endIcon}
      startIcon={startIcon}
      message="Pending Transactions"
      onPress={onPress}
    />
  )
}

export default PendingTransactions
