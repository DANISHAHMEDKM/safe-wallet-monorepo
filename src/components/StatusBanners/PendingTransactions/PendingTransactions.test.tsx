import { render } from '@/src/tests/test-utils'
import PendingTransactions from '.'

describe('PendingTransactions', () => {
  it('should render the default markup', () => {
    const container = render(<PendingTransactions number={'2'} onPress={jest.fn()} />)

    expect(container).toMatchSnapshot()
  })

  it('should render the pending transactions in fullWidth layout', () => {
    const container = render(<PendingTransactions number={'2'} fullWidth onPress={jest.fn()} />)

    expect(container).toMatchSnapshot()
  })
})
