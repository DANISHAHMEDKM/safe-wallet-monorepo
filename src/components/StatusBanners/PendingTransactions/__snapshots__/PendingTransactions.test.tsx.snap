// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingTransactions should render the default markup 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={false}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff0e0",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "flexDirection": "row",
            "gap": 12,
            "justifyContent": "center",
            "paddingBottom": 12,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 12,
            "width": "100%",
          }
        }
      >
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FF8C00",
                "borderBottomLeftRadius": 100000,
                "borderBottomRightRadius": 100000,
                "borderTopLeftRadius": 100000,
                "borderTopRightRadius": 100000,
                "flexDirection": "column",
                "height": 28,
                "justifyContent": "center",
                "maxHeight": 28,
                "maxWidth": 28,
                "minHeight": 28,
                "minWidth": 28,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
                "width": 28,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff0e0",
                  "fontSize": 14,
                }
              }
              suppressHighlighting={true}
            >
              2
            </Text>
          </View>
        </View>
        <Text
          style={
            {
              "color": "#FF8C00",
              "fontFamily": "DMSans-SemiBold",
              "fontSize": 16,
            }
          }
          suppressHighlighting={true}
        >
          Pending Transactions
        </Text>
        <View>
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#FF8C00",
                  "fontSize": 20,
                },
                undefined,
                {
                  "fontFamily": "SafeIcons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNGestureHandlerButton>
`;

exports[`PendingTransactions should render the pending transactions in fullWidth layout 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={false}
  exclusive={true}
  handlerTag={2}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff0e0",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "flexDirection": "row",
            "gap": 12,
            "justifyContent": "center",
            "paddingBottom": 12,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 12,
            "width": "100%",
          }
        }
      >
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FF8C00",
                "borderBottomLeftRadius": 100000,
                "borderBottomRightRadius": 100000,
                "borderTopLeftRadius": 100000,
                "borderTopRightRadius": 100000,
                "flexDirection": "column",
                "height": 28,
                "justifyContent": "center",
                "maxHeight": 28,
                "maxWidth": 28,
                "minHeight": 28,
                "minWidth": 28,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
                "width": 28,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff0e0",
                  "fontSize": 14,
                }
              }
              suppressHighlighting={true}
            >
              2
            </Text>
          </View>
        </View>
        <Text
          style={
            {
              "color": "#FF8C00",
              "fontFamily": "DMSans-SemiBold",
              "fontSize": 16,
            }
          }
          suppressHighlighting={true}
        >
          Pending Transactions
        </Text>
        <View>
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#FF8C00",
                  "fontSize": 20,
                },
                undefined,
                {
                  "fontFamily": "SafeIcons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNGestureHandlerButton>
`;
